File: DataSets/DataSet-New/LeetCodeTasks - Debug/Fixed/3111_minimum-rectangles-to-cover-points/solutions/python3/solution.py
from typing import List

class Solution:
    def minRectanglesToCoverPoints(self, points: List[List[int]], w: int) -> int:
        # Sorting points based on the x-coordinate
        points.sort()

        rectangles = 0
        i = 0
        n = len(points)

        while i < n:
            rectangles += 1  # We are going to use a new rectangle
            # The starting x of the rectangle
            start_x = points[i][0]
            # The maximum x for this rectangle
            max_x = start_x + w
            
            # Move to the next point that is out of the current rectangle's width
            while i < n and points[i][0] <= max_x:
                i += 1

        return rectangles

File: DataSets/DataSet-New/LeetCodeTasks - Debug/Fixed/3111_minimum-rectangles-to-cover-points/solutions/cpp/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int minRectanglesToCoverPoints(std::vector<std::vector<int>>& points, int w) {
        // Sort points based on the x-coordinate
        std::sort(points.begin(), points.end());

        int count = 0;
        int n = points.size();
        int i = 0;

        while (i < n) {
            count++;
            int x_start = points[i][0];
            int x_end = x_start + w;

            // Move to the next point outside the current rectangle range
            while (i < n && points[i][0] <= x_end) {
                i++;
            }
        }

        return count;
    }
};
